torch/csrc/tensor/python_tensor.cpp

```diff
@@ -434,27 +419,29 @@ static void py_bind_tensor_types(
   }
 }
 
-static bool PyTensorType_Check(PyObject* obj) {
-  auto it = std::find_if(
-      tensor_types.begin(), tensor_types.end(), [obj](PyTensorType* x) {
-        return (PyObject*)x == obj;
-      });
-  return it != tensor_types.end();
-}
-
 void py_set_default_tensor_type(PyObject* obj) {
```

Comment:
Most of everything in this PR is non-objectionable, but the changes here seem a bit weird. What's going on?

Reply:
PyTensorType_Check only checks object whether is in tensor_types, and it is assumed that the data stored on the torch._tensor_classes and tensor_types vector are consistent.
Now tensor_types are only used for CPU and GPU tensor types, which is a local static vector. But _tensor_classes is python side, which can be registered from other libraries for custom device tensor types. So from checked object based on local tensor_types vector to global torch._tensor_classes.

Reply:
Drop this bit from this PR, if you really want it let's litigate it on its own. But this is the same "I don't want to aid and abet more usage of the legacy constructor style"

Reply:
@ezyang ok, Dropped this bit in this PR, and i will create a issue for legacy constructor style.